### PR TITLE
SDP-977 - Multi-tenant Docker Compose

### PR DESCRIPTION
**Pull request by marwen-abid**
## Changes 
- Add new environment variable `ENABLE_SCHEDULER`  (default: false): This flag enables the scheduler. Scheduler will remain disabled until work on Kafka is completed. 
- Add new environment variable `ENABLE_MULTITENANT_DB` (default: true): When this flag is enabled, we use the Multi-tenant connection pool with router for the API. When it's disabled, we use the single tenant connection pool. 
- Set `CORS_ALLOWED_ORIGINS: "*"` in docker-compose. This is needed until we fix SDP-980
- Use `multitenant-alpha` frontend-image for docker-compose. 
- Disable TSS for docker-compose. This will remain disabled until work on Kafka is completed. 
- `main.sh` add scripts for creating example tenants during local development. 
- Disable `e2e tests` when PR target is `sdp-multitenant`. This will remain disabled until Kafka work is completed.

### CODE REVIEW

File: cmd/serve.go [hunk 1]
```diff
@@ -408,13 +424,17 @@ func (c *ServeCommand) Command(serverService ServerServiceInterface, monitorServ
...
 			serveOpts.AnchorPlatformAPIService = apAPIService
 
-			// Starting Scheduler Service (background job)
-			log.Ctx(ctx).Info("Starting Scheduler Service...")
-			schedulerJobRegistrars, err := serverService.GetSchedulerJobRegistrars(ctx, serveOpts, schedulerOptions, apAPIService)
-			if err != nil {
-				log.Ctx(ctx).Fatalf("Error getting scheduler job registrars: %v", err)
+			// Starting Scheduler Service (background job) if enabled
+			if serveOpts.EnableScheduler {
+				log.Ctx(ctx).Info("Starting Scheduler Service...")
```
Comment (reecexlm) [inline cmd/serve.go:429]: wanna add some additional debugging info?  eg some of the serveOpts props?

File: dev/main.sh [hunk 2]
```diff
@@ -12,8 +12,56 @@ echo "====> ✅Step 1: finish preparation"
...
 echo $DIVIDER
 echo "====> 👀Step 2: start calling docker compose up"
-docker-compose -f docker-compose-sdp-anchor.yml down && docker-compose -f docker-compose-sdp-anchor.yml up --abort-on-container-exit
+docker-compose down && docker-compose -p sdp-multi-tenant up -d --build
 echo "====> ✅Step 2: finish calling docker-compose up"
 
+# Initialize tenants
 echo $DIVIDER
-echo "🎉🎉🎉🎉 SUCCESS! 🎉🎉🎉🎉"
+echo "====> 👀Step 3: initialize tenants"
```
Comment (reecexlm) [inline dev/main.sh:20]: do we wanna do something like
if ! command -v curl &> /dev/null; then
    echo "Error: curl is not installed." >&2
    exit 1
fi
at the beginnign of the script?

**Review by reecexlm — APPROVED** [review]
left couple comments but looks good

File: cmd/serve.go (hunk 1 above)
Comment (marwen-abid) [reply cmd/serve.go:429]: Individual jobs do their own logging. The only other relevant Opts is `MaxInvitationSMSResendAttempts`, but that gets printed when relevant in `SendReceiverWalletInviteService`

File: dev/main.sh (hunk 2 above)
Comment (marwen-abid) [reply dev/main.sh:20]: Addressed, thank you!

File: internal/serve/serve.go [hunk 3]
```diff
@@ -99,16 +101,29 @@ func (opts *ServeOptions) SetupDependencies() error {
...
 
-	// Setup Database:
+	// Setup Tenant Router & Manager
 	dbConnectionPool, err := db.OpenDBConnectionPoolWithMetrics(opts.DatabaseDSN, opts.MonitorService)
 	if err != nil {
 		return fmt.Errorf("error connecting to the database: %w", err)
 	}
-	opts.Models, err = data.NewModels(dbConnectionPool)
+	opts.tenantManager = tenant.NewManager(tenant.WithDatabase(dbConnectionPool))
+	opts.tenantRouter = router.NewMultiTenantDataSourceRouter(opts.tenantManager)
```
[the remark this answers is not in the document]
Comment (marwen-abid) [reply internal/serve/serve.go]: thank you! Addressed in c4cd369ee814659eed630af4aff7d75e081da646

**Pull request merged**